### PR TITLE
Rename oltp_endpoint to otlp_endpoint to match opentelemetry spec and lib name

### LINKED ISSUE #5067: Configuration Key For The Open Telemetry Endpoint Is Inconsistent with OpenTelemetry Spec
OpenTelemetry spec is here: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md

OpenTelemetry Protocol is shortened to OTLP according to their documentation and most libs I have seen.

The config for tracing for cortex however is inconsistent with this and instead uses:

`oltp_endpoint`

I know this may seem like a Mickey Mouse change, however if you work with opentelemetry extensively you might be in for a bad time when you try to set this like every other implementation.

My proposal is to rename this configuration key to `otlp_endpoint` for consistency sake.

### PR DESCRIPTION
**Pull request by sharkymcdongles**
**What this PR does**:
Renames oltp_endpoint to otlp_endpoint to match opentelemetry spec and lib name

**Which issue(s) this PR fixes**:
Fixes #5067 

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by yeya24 — APPROVED** [review]
Thanks for this change. It makes sense to me.

**Comment by alanprot** [comment]
I also agree.. thanks for pointing this out… I just wonder if there is a way to keep both configs for some time (maybe hiding from the docs and giving a warn on the logs) so we don’t break customers that are already using this config

**Review by friedrichg — DISMISSED** [review]
Thanks!
@alanprot To me otlp support is experimental, we need to put it in v1.x Guarantees as we do with all new risky features.

**Comment by yeya24** [comment]
I agree this feature is just experimental and we shoulbe be able to change it.

**Comment by alanprot** [comment]
> I agree this feature is just experimental and we shoulbe be able to change it.

Its not marked as experimental though, right?

But iI'm ok either way i guess...

**Comment by yeya24** [comment]
Maybe we should follow the doc and make sure we are backward compatible.

**Comment by friedrichg** [comment]
> Its not marked as experimental though, right?

No, it's not experimental. That was just a suggestion.

> Maybe we should follow the doc and make sure we are backward compatible.

yes, that is the way forward. fyi @sharkymcdongles

**Comment by sharkymcdongles** [comment]
@friedrichg I will make changes then this week. Cheers!

**Comment by sharkymcdongles** [comment]
@friedrichg I just made some adjustments hope it is okay. Was a quick and dirty during a too long meeting where I was bored af and wanted something to distract me.

**Comment by sharkymcdongles** [comment]
Whoopsie daisies, I forgot I am working with Golang and not Python and did some stupid stuff. Fixed.

File: docs/configuration/config-file-reference.md [hunk 1]
```diff
@@ -4050,10 +4050,16 @@ The `tracing_config` configures backends cortex uses.
 [type: <string> | default = "jaeger"]
 
 otel:
+
+  # DEPRECATED: use otel.otlp_endpoint instead
   # otl collector endpoint that the driver will use to send spans.
```
Comment (friedrich-at-adobe) [inline docs/configuration/config-file-reference.md:4055]:
```suggestion
  # DEPRECATED: use otel.otlp_endpoint instead.
```

Comment (sharkymcdongles) [reply docs/configuration/config-file-reference.md]: Added the suggested change. Thanks!

**Comment by sharkymcdongles** [comment]
Not really sure what is failing now. Anyone able to help? I didn't touch the compactor, and my documentation looks proper.

**Comment by sharkymcdongles** [comment]
NM seems it was the newline. Not sure about the compactor test though.

File: pkg/tracing/tracing.go [hunk 2]
```diff
@@ -83,8 +88,19 @@ func SetupTracing(ctx context.Context, name string, c Config) (func(context.Cont
...
-			otlptracegrpc.WithEndpoint(c.Otel.OltpEndpoint),
+			otlptracegrpc.WithEndpoint(c.Otel.OtlpEndpoint),
+		}
+
+		if (c.Otel.OtlpEndpoint == "") && (len(c.Otel.OltpEndpoint) > 0) {
+			level.Warn(util_log.Logger).Log("msg", "DEPRECATED: otel.oltp is deprecated use otel.otlp")
+			options = []otlptracegrpc.Option{
+				otlptracegrpc.WithEndpoint(c.Otel.OltpEndpoint),
+			}
 		}
```
[the remark this answers is not in the document]
Comment (sharkymcdongles) [reply pkg/tracing/tracing.go:104]: I already did that here: [91](https://github.com/cortexproject/cortex/blob/85fde3799f769c41922b7108387e6f7b3e5a6b8e/pkg/tracing/tracing.go#L91
)

**Logic is:**

If both are set log warning that both are set and otel.oltp will be removed and set options to use OtlpEndpoint.

If only OltpEndpoint is set then use it and warn as well.

Then once it is fully removed then both the if from 91 and 99 can be removed.

Maybe there is a cleaner way than this if else condition though not sure.

File: pkg/tracing/tracing.go [hunk 3]
```diff
@@ -39,6 +39,7 @@ type Config struct {
 
 type Otel struct {
 	OltpEndpoint   string              `yaml:"oltp_endpoint" json:"oltp_endpoint"`
```
[the remark this answers is not in the document]
Comment (sharkymcdongles) [reply pkg/tracing/tracing.go:41]: Done.

File: docs/configuration/config-file-reference.md [hunk 4]
```diff
@@ -4050,10 +4050,14 @@ The `tracing_config` configures backends cortex uses.
 [type: <string> | default = "jaeger"]
 
 otel:
-  # otl collector endpoint that the driver will use to send spans.
+  # DEPRECATED: use otel.otlp_endpoint instead.
   # CLI flag: -tracing.otel.oltp-endpoint
   [oltp_endpoint: <string> | default = ""]
```
Comment (friedrichg) [inline docs/configuration/config-file-reference.md:4056]:
```suggestion
```
to pass the lint, because hidden was added

**Comment by yeya24** [comment]
> --- FAIL: TestCompactor_ShouldCompactOnlyShardsOwnedByTheInstanceOnShardingEnabledWithShuffleShardingAndMultipleInstancesRunning (4.72s)
    compactor_test.go:1298: 
        	Error Trace:	/__w/cortex/cortex/pkg/compactor/compactor_test.go:1298
        	Error:      	Should be true
        	Test:       	TestCompactor_ShouldCompactOnlyShardsOwnedByTheInstanceOnShardingEnabledWithShuffleShardingAndMultipleInstancesRunning
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x190b629]

This is a flaky test we need to take care of.

**Pull request merged**